Packages/ClientRuntime/Sources/Waiters/WaiterUtils.swift

```diff
@@ -0,0 +1,120 @@
+    /// - Parameter int: The `Int` to represent as a JMESPath number, or as a JMESPath `null` if `int` is nil.
+    public init(_ int: Int?) {
+        guard let int = int else { self = .null; return }
+        self = .number(Double(int))
+    }
+
+    /// Creates a `JMESValue` from a Swift `Double`.
+    /// - Parameter double: The `Double` to represent as a JMESPath number, or as a JMESPath `null` if `double` is nil.
+    public init(_ double: Double?) {
+        guard let double = double else { self = .null; return }
+        self = .number(double)
+    }
+
+    /// Creates a `JMESValue` from a Swift `Bool`.
+    /// - Parameter bool: The `Bool` to represent as a JMESPath Boolean, or as a JMESPath `null` if `bool` is nil.
+    public init(_ bool: Bool?) {
+        guard let bool = bool else { self = .null; return }
+        self = .boolean(bool)
+    }
+
+    /// Creates a `JMESValue` from a Swift `String`.
+    /// - Parameter string: The `String` to represent as a JMESPath String, or as a JMESPath `null` if `string` is nil.
+    public init(_ string: String?) {
+        guard let string = string else { self = .null; return }
+        self = .string(string)
+    }
+
+    /// Creates a `JMESValue` from a type that is raw-representable by a string.
+    /// Typically this will be an enumeration of a limited set of expected string values.
+    /// The resulting value will be a string for purposes of comparing it to other values.
+    /// - Parameter rr: The `RawRepresentable` to represent as a JMESPath String, or as a JMESPath `null` if `rr` is nil.
+    public init<T: RawRepresentable>(_ rr: T?) where T.RawValue == String {
+        guard let string = rr?.rawValue else { self = .null; return }
+        self = .string(string)
+    }
+
+    /// Compares two `JMESValue` values, testing for equality per the JMESPath rules.
+    /// - Parameters:
+    ///   - lhs: The left value
+    ///   - rhs: The right value
+    /// - Returns: The result of the Swift comparison of the associated values if `lhs` and `rhs` are the same JMESPath type, or `false` otherwise.
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        switch (lhs, rhs) {
+        case (.number(let left), .number(let right)):
+            return left == right
+        case (.boolean(let left), .boolean(let right)):
+            return left == right
+        case (.string(let left), .string(let right)):
+            return left == right
+        case (.null, .null):
+            return true
+        default:
+            return false
+        }
+    }
+
+    /// Compares two `JMESValue` values, testing for order per the JMESPath rules.
+    /// Comparison fails if both left and right are not JMESPath numbers.
+    /// - Parameters:
+    ///   - lhs: The left value
+    ///   - rhs: The right value
+    /// - Returns: The result of the Swift comparison of the associated values if `lhs` and `rhs` are both JMESPath numbers, or `false` otherwise.
```

Comment:
```suggestion
    /// Compares two `JMESValue` values per the JMESPath rules.
    /// This always returns `false` if both the left and right values are not JMESPath numbers.
    /// - Parameters:
    ///   - lhs: The left value
    ///   - rhs: The right value
    /// - Returns: `true`  if `lhs` and `rhs` are both JMESPath numbers and the `lhs` value is less than the `rhs` value, otherwise returns `false`.
```